File: src/swagger/Path.php
<?php

namespace Ething\Swagger;



class Path {
	
	private $path;
	private $data;
	private $parent;
	
	public function __construct($path, $data, $parent) {
		
		$this->parent = $parent;
		$this->path = $path;
		$this->data = $data;
		
	}
	
	
	public function path() {
		return $this->path;
	}
	
	public function url() {
		return $this->parent->url() . $this->path;
	}
	
	public function swagger() {
		return $this->parent;
	}
	
	public function parameters() {
		
		if(!isset($this->paramters_)){
			$parameters = array();
			
			if(!empty($this->data->parameters)) {
				foreach($this->data->parameters as $param){
					$parameters[] = $param->in == 'body' ? new BodyParameter($this, $param) : new StandardParameter($this, $param);
				}
			}
			
			$this->paramters_ = $parameters;
		}
		
		return $this->paramters_;
	}
	
	public function operations() {
		if(!isset($this->operations_)){
			$operations = array();
			
			foreach($this->data as $method => $opdata){
				if(in_array($method, array('get','put','post','delete','options','head','patch'))){
					$operations[] = new Operation($method, $opdata, $this);
				}
			}
			
			$this->operations_ = $operations;
		}
		return $this->operations_;
	}
	
}

File: ething/swagger/Path.py
# coding: utf-8


from future.utils import iteritems
from .Parameter import instanciate as instanciate_parameter
from .Operation import Operation

class Path(object):
    
    
    def __init__ (self, pathname, data, parent):
        
        self.parent = parent;
        self.__pathname = pathname;
        self.data = data;
        self.__parameters = None;
        self.__operations = None
    
    
    @property
    def pathname (self):
        return self.__pathname
    
    @property
    def url (self):
        return self.parent.url + self.pathname
    
    @property
    def root (self):
        return self.parent;
    
    @property
    def parameters (self):
        
        if self.__parameters is None:
            self.__parameters = []
            
            for paramobj in self.data.get('parameters', []):
                self.__parameters.append(instanciate_parameter(paramobj, self))
            
        return self.__parameters
    
    @property
    def operations (self):
        if self.__operations is None:
            self.__operations = []
            
            for method, operationobj in iteritems(self.data):
                if method.lower() in ['get','put','post','delete','options','head','patch']:
                    self.__operations.append(Operation(method, operationobj, self))
        
        return self.__operations
    
    

